### ISSUE TITLE
Breaking changes in version 2

### ISSUE THREAD

**Issue by sa-he** [anakic/Jot#29]
What happened, that version 2 introduced big breaking changes?

I've been building my project on Jot's TrackableAttribute. How whould I update to version 2 with as little intrusion as possible? Also it would be good if the settings that are on my users computers would not be lost.

My current version 1.4.1 approach:

```C#
// in ctor of base view model (WPF & MVVM)
Configuration = stateTracker.Configure(this)
    .IdentifyAs("static")
    .SetAutoPersistEnabled(true); // persist at application shutdown

Configuration.StatePersisted += Configuration_StatePersisted;
Configuration.StateApplied += Configuration_StateApplied;
```

```C#
protected virtual void Configuration_StateApplied(object sender, EventArgs e) 
{ 
    // Enable child class to react on this event
}
protected virtual void Configuration_StatePersisted(object sender, EventArgs e) 
{
    // Enable child class to react on this event
}
```

```C#
// every child view model marks relevant properties with TrackableAttribute 
// and has it automatically persisted and restored
[Trackable]
public string Property1 { get; set; }
```
The files I'm ending up with are:
%appdata%\myproject\MyViewModel_static.json

```JSON
[
  {
    "Type": "System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089",
    "Name": "Property1",
    "Value": "foo"
  }
]
```

**Comment by anakic** [comment]
@SchwabenProgger yeah, it's a bit of an overhaul but it was overdue.

TLDR: The reasons for the update are supporting .net standard and some improvements to the API and the inner workings.

Here's a list of changes and reasons:
1. The library now targets netstandard so it can be used from .net core. Version 3.0 of .net core is coming out next week and it will support WinForms and WPF. Also, it seems likely that Jot will be useful in web situations in .net core (Blazor).
1. The API has been improved in a few places, e.g. instead of calling `cfg.AddProperties<Form>(f=> f.Width, f => f.Height)`, now you can call `cfg.Properties(f => new { f.Width, f.Height })`.
1. Tracked properties can now drill into other objects, e.g.  `tracker.Configure<MyForm>().Properties(f => new { f.Width, f.Height, f.tabControl1.SelectedIndex /*drilling into tabControl1*/ })`
1. The new version simplifies storage logic (`IStoreFactory` was removed, `IStore` was simplified) so it's easier to introduce new storage options. 
1. The new version removes the `IConfigurationInitializer` interface. This interface was intended to allow configuring tracking per-type, but it was messy. Now, tracking is configured per-type by design, so the `IConfigurationInitializer` and related code is no longer needed. 
1. Events in the new version are replaced with callbacks. This is a cosmetic change to enable method chaining. E.g. instead of `StateApplied += Configuration_StatePersisted`, you would now call `WhenAppliedState(Configuration_StatePersisted)`.
1. I've removed the `[Trackable]` and `[TrackingKey]` attributes temporarily. I wasn't sure if they're needed, because they're less flexible than implementing `ITrackingAware` or calling the configuration methods so I left them out initially. However, being able to slap a `[Trackable]` attribute on a property is simpler than including that property when calling the `Properties` method, so it does seem to be a good idea to bring them back. 

All in all, I'm really happy with the new version of Jot. It's cleaner and can do a more (e.g. drill into nested properties). 

In your case, the following code should do it:

```csharp
// set up tracking for TheVM type (tracking is configured per-type):
tracker.Configure<TheVM>()
   .Id(vm => "static" /*or e.g. vm.GetType().Name + "static"*/)
   .Properties(vm => new { vm.Property1 /*, vm.Prop2, vm.Prop3... */ })
   .WhenAppliedState(vm => { /**/ })
   .WhenPersisted(vm => { /**/ })

// track the vm instance
tracker.Track(vmInstance);
```
The settings on your client machines will not be lost, the changes did not affect the structure or location of the json file.
